src/posix/platform/netif.cpp

```diff
@@ -1598,41 +1602,62 @@ static void processMLDEvent(otInstance *aInstance)
 
 #if defined(__linux__)
 // set up the tun device
-static void platformConfigureTunDevice(const char *aInterfaceName, char *deviceName, size_t deviceNameLen)
+static void platformConfigureTunDevice(otPlatformConfig *aPlatformConfig)
 {
     struct ifreq ifr;
+    const char *interfaceName;
 
     sTunFd = open(OPENTHREAD_POSIX_TUN_DEVICE, O_RDWR | O_CLOEXEC | O_NONBLOCK);
     VerifyOrDie(sTunFd >= 0, OT_EXIT_ERROR_ERRNO);
 
     memset(&ifr, 0, sizeof(ifr));
-    ifr.ifr_flags = IFF_TUN | IFF_NO_PI | static_cast<short>(IFF_TUN_EXCL);
+    ifr.ifr_flags = IFF_TUN | IFF_NO_PI;
+    if (!aPlatformConfig->mPersistentInterface)
+    {
+        ifr.ifr_flags |= static_cast<short>(IFF_TUN_EXCL);
+    }
 
-    if (aInterfaceName)
+    interfaceName = aPlatformConfig->mInterfaceName;
```

Comment:
What happens if `--persistent-interface` is specified but interface name is not given?
Is it a meaningful usecase?

Reply:
In this case, it work as same as before, a device with name "wpanX" will be created, then we make it persistent.

Reply:
What happens if we run `-I wpan0 --persistent-interface` first, and then run `-I wpan0`?

Can the second run be successfully, and the wpan0 interface be removed when the process quit?

Reply:
No, the second run fails, due to not able to create `wpan0`

Reply:
Should we make sure that the second run can be successful?

For example, we may enable `--persistent-interface` on one product, and later decide to disable `--persistent-interface`. Current implementation would not allow simply removing `--persistent-interface` to work.

Reply:
I requires a reboot to update the firmware, right ? after a reboot, the device will be removed.